client/src/effects/filter.effects.js

```diff
@@ -1,102 +1,108 @@
+      reqs = ["Citizen"];
       return reqs;
     }
+    reqs = ["Citizen", "Permanent Residency", "International"];
+    return reqs;
   }
+}
 
-  function parseCoopReqs(coOp) {
-    if (coOp === "Yes") {
-      return true;
-    } else {
-      return false;
-    }
+function parseCoopReqs(coOp) {
+  if (coOp === "Yes") {
+    return true;
+  } else {
+    return false;
   }
+}
 
-  function parseLanguages(languages) {
-    let editedLanguages = [];
-    for (let i = 0; i < languages.length; i++) {
-      if (languages[i] === "C++") {
-        editedLanguages.push("Cpp");
-        editedLanguages.push("C++");
-      } else if (languages[i] === "Cpp") {
-        editedLanguages.push("C++");
-        editedLanguages.push("Cpp");
-      } else {
-        editedLanguages.push(languages[i]);
-      }
+function parseLanguages(languages) {
+  let editedLanguages = [];
+  for (let i = 0; i < languages.length; i++) {
+    if (languages[i] === "C++") {
+      editedLanguages.push("Cpp");
+      editedLanguages.push("C++");
+    } else if (languages[i] === "Cpp") {
+      editedLanguages.push("C++");
+      editedLanguages.push("Cpp");
+    } else {
+      editedLanguages.push(languages[i]);
     }
-    return editedLanguages;
   }
+  return editedLanguages;
+}
 
-  function createJobObject(jobPosting) {
-    // parsing functions
-    const experience = parseExperience(jobPosting.requirements.experience);
-    const candidates = parseCitizenshipReqs(jobPosting.details.candidates);
-    const coOp = parseCoopReqs(jobPosting.details.coOp);
-    const concepts = parseConcepts(jobPosting.requirements.concepts);
-    const languages = parseLanguages(jobPosting.requirements.languages);
+function createJobObject(jobPosting) {
+  // parsing functions
+  const experience = parseExperience(jobPosting.requirements.experience);
+  const candidates = parseCitizenshipReqs(jobPosting.details.candidates);
+  const coOp = parseCoopReqs(jobPosting.details.coOp);
+  const concepts = parseConcepts(jobPosting.requirements.concepts);
+  const languages = parseLanguages(jobPosting.requirements.languages);
 
-    return {
-      jobId: jobPosting.jobId,
-      experience: experience,
-      gpa: jobPosting.requirements.gpa,
-      gpaValue: jobPosting.requirements.gpaValue,
-      languages: languages,
-      frameworks: jobPosting.requirements.frameworks,
-      tools: jobPosting.requirements.tools,
-      concepts: concepts,
-      candidates: candidates,
-      academicReq: jobPosting.details.academicReq,
-      coOp: coOp,
-    };
-  }
+  return {
+    jobId: jobPosting.jobId,
+    experience: experience,
+    gpa: jobPosting.requirements.gpa,
+    gpaValue: jobPosting.requirements.gpaValue,
+    languages: languages,
+    frameworks: jobPosting.requirements.frameworks,
+    tools: jobPosting.requirements.tools,
+    concepts: concepts,
+    candidates: candidates,
+    academicReq: jobPosting.details.academicReq,
+    coOp: coOp,
+  };
+}
```

Comment:
does this object need to get added with the other attributes?

Reply:
Can you clarify?

Reply:
do you need the notes, matches, profile...etc? Just wondering what this object is used for and if it requires all the job object information or not

Reply:
or is it ok to just maintain these attributes and not have all the attributes required for a job object?